### PR TITLE
Applicants understand what to do with the code

### PR DESCRIPTION
**Pull request by dsamojlenko**
# What this does
Updates the Invite screen according to design [changes described here](https://trello.com/c/JGD3YPhx/500-applicants-understand-what-to-do-with-the-code).

## To test
- Start a new application
- Complete the Personal and Consent forms
- See the new Invite screen

### CODE REVIEW

**Review by CalvinRodo — COMMENTED** [review]
Looks like the design. 🐈

**Review by CalvinRodo — DISMISSED** [review]
That last comment should have been an approve

File: config/locales/en.json [hunk 1]
```diff
@@ -504,11 +504,18 @@
 	"dashboard.overall-health": "Patient's Overall Health",
 	"dashboard.expedited-processing": "Expedited processing for terminal and grave conditions",
 	"conditions.incomplete": "Not Completed",
+	"invite.expires": "This code will expire on %s",
```
Comment (keithiopia) [inline config/locales/en.json]: Missing a full stop/period on the end of the sentence.

File: config/locales/en.json [hunk 2]
```diff
@@ -504,11 +504,18 @@
...
+	"invite.code_loss_instruction": "If you lose your code, or it expires, fill out this form again to generate another code",
```
Comment (keithiopia) [inline config/locales/en.json]: Missing a full stop/period on the end of the sentence.

File: config/locales/en.json [hunk 3]
```diff
@@ -504,11 +504,18 @@
...
 	"invite.title": "Form submitted",
-	"invite.give_code": "To finish your application, give this code and link to your doctor or nurse practioner:",
+	"invite.give_code": "To finish your medical report, give this code to your doctor or nurse practitioner.",
```
Comment (keithiopia) [inline config/locales/en.json]: Is it possible to give this text more emphasis? The standard font-weight looks a bit high globally (to me). What font-weight do we use for bold?

File: config/locales/en.json [hunk 4]
```diff
@@ -504,11 +504,18 @@
...
+	"invite.give_code_details": "They will use this code to fill out their part of the medical report form, and send it to us at Service Canada.",
 	"invite.medical_professional_instruction": "Your doctor or nurse practioner will use this unique code to fill out a report about your medical condition(s), and how they impact your ability to work.",
 	"invite.print_button": "Print this code",
 	"invite.url": "https://cppd-medical-report-stage.herokuapp.com/en/doctor",
+	"invite.for_medical_professionals": "For health professionals:",
```
Comment (keithiopia) [inline config/locales/en.json]: Can we put this in italics and maybe add a little more whitespace between this line and the box aimed at the applicant?
I want there to be more of gap between them to help emphasize the top bit is for the applicant, and this chunk of steps is for someone else.

File: views/pages/invite.njk [hunk 5]
```diff
@@ -4,12 +4,21 @@
     <h1>{{ __('invite.title') }}</h1>
```
Comment (keithiopia) [inline views/pages/invite.njk]: The page title should be `Give this code to your doctor or nurse practitioner`.
We want to remove anything that could get in the way of our primary instruction to the applicant.

File: views/pages/invite.njk [hunk 6]
```diff
@@ -4,12 +4,21 @@
...
 
     <div class="w-4/5 block border border-gray-500 border-solid p-5">
-        <p>{{ __('invite.give_code') }}</p>
+        <p class="font-semi-bold">{{ __('invite.give_code') }}</p>
+        <p>{{ __('invite.give_code_details') }}</p>
         <p class="font-bold text-6xl">{{ applicationCode }}</p>
-        <!-- TODO Dynamically generate the URL, rather than pulling in from translation -->
-        <p class="font-bold">{{ medicalProfessionalUrl }}</p>
-        <p>{{ __('invite.medical_professional_instruction') }}</p>
+        <p>{{ __('invite.expires', '[PLACEHOLDER DATE]') }}</p>
```
Comment (keithiopia) [inline views/pages/invite.njk]: Can we move line 10 and 11 below the button?
It's secondary information.

File: views/pages/invite.njk [hunk 7]
```diff
@@ -4,12 +4,21 @@
     <h1>{{ __('invite.title') }}</h1>
 
     <div class="w-4/5 block border border-gray-500 border-solid p-5">
-        <p>{{ __('invite.give_code') }}</p>
+        <p class="font-semi-bold">{{ __('invite.give_code') }}</p>
+        <p>{{ __('invite.give_code_details') }}</p>
         <p class="font-bold text-6xl">{{ applicationCode }}</p>
```
Comment (keithiopia) [inline views/pages/invite.njk]: Can we move the code to the top of this div box? It's the most important piece of information/data on this page, and follows on from the new page heading.

File: config/locales/en.json [hunk 8]
```diff
@@ -504,11 +504,18 @@
...
+	"invite.go_to": "Go to %s",
+	"invite.enter_the_code": "Enter the code %s and your patient's date of birth.",
```
Comment (keithiopia) [inline config/locales/en.json]:
```suggestion
	"invite.enter_the_code": "Enter the code '%s' and your patient's date of birth.",
```
Assuming you can put that placeholder tag in inverted commas without breaking anything.

**Review by keithiopia — COMMENTED** [review]
Comments left.

Ignoring the no consent path at this point.

File: config/locales/en.json (hunk 1 above)
Comment (dsamojlenko) [reply config/locales/en.json]: done

File: config/locales/en.json (hunk 2 above)
Comment (dsamojlenko) [reply config/locales/en.json]: done

File: config/locales/en.json (hunk 3 above)
Comment (dsamojlenko) [reply config/locales/en.json]: done. had just mis-spelled the font-semibold class.

File: config/locales/en.json (hunk 4 above)
Comment (dsamojlenko) [reply config/locales/en.json]: done

File: views/pages/invite.njk (hunk 5 above)
Comment (dsamojlenko) [reply views/pages/invite.njk]: done

File: views/pages/invite.njk (hunk 6 above)
Comment (dsamojlenko) [reply views/pages/invite.njk]: done

File: views/pages/invite.njk (hunk 7 above)
Comment (dsamojlenko) [reply views/pages/invite.njk]: done

File: config/locales/en.json (hunk 8 above)
Comment (dsamojlenko) [reply config/locales/en.json]: indeed

File: api/controllers/InviteController.js [hunk 9]
```diff
@@ -16,8 +16,8 @@ module.exports = {
     req.session.destroy();
 
     // Get full URL
-    const protocol = req.connection.encrypted? 'https' : 'http';
-    const medicalProfessionalRoute = sails.config.routes['GET /en/medical-professional'].i18n[req.i18n.locale];
+    const protocol = req.connection.encrypted ? 'https' : 'http';
+    const medicalProfessionalRoute = sails.route('medical-professional');
     let medicalProfessionalUrl = protocol + '://' + req.headers.host + medicalProfessionalRoute;
```
Comment (jeffmaher) [inline api/controllers/InviteController.js]: Suggestion for part for removing placeholder for valid date.

```suggestion
    let medicalProfessionalUrl = protocol + '://' + req.headers.host + medicalProfessionalRoute;
    
    // Get timeframe for when the consent expires
    let submissionMoment = moment(medicalReport.applicantSubmittedAt);
    let validTil = submissionMoment.add(3, 'y').format('LL');
```

Comment (jeffmaher) [reply api/controllers/InviteController.js]: You'll also need to add to the top:

`const moment = require('moment');` 

and inject into the view data dictionary.

File: views/pages/invite.njk [hunk 10]
```diff
@@ -5,11 +5,23 @@
...
-            <p>{{ __('invite.give_code') }}</p>
             <p class="font-bold text-6xl">{{ applicationCode }}</p>
-            <p class="font-bold">{{ medicalProfessionalUrl }}</p>
-            <p>{{ __('invite.medical_professional_instruction') }}</p>
+            
+            <p class="font-semibold">{{ __('invite.give_code') }}</p>
+            <p>{{ __('invite.give_code_details') }}</p>
+
             <a href="javascript:window.print()" class="button-link w-2/5">{{ __('invite.print_button') }}</a>
+            <p class="mt-8">{{ __('invite.expires', '[PLACEHOLDER DATE]') }}</p>
```
Comment (jeffmaher) [inline views/pages/invite.njk]: Will add suggested code for this part.

**Review by keithiopia — DISMISSED** [review]
I think @jeffmaher is working on the placeholder date bit, otherwise looks good to go!

File: api/controllers/InviteController.js (hunk 9 above)
Comment (dsamojlenko) [reply api/controllers/InviteController.js]: Merci, I added all of this!

**Review by jeffmaher — APPROVED** [review]
LGTM 🍕

**Pull request merged**